test/jaxp/javax/xml/jaxp/unittest/xpath/XPathExpFollowingTest.java

```diff
@@ -0,0 +1,154 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package xpath;
+
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+/*
+ * @test
+ * @bug 8289510
+ * @library /javax/xml/jaxp/unittest
+ * @run testng/othervm xpath.XPathExpFollowingTest
+ * @summary Tests for XPath following/following-sibling axis specifier.
+ */
+public class XPathExpFollowingTest extends  XPathTestBase {
+    /*
+     * DataProvider: provides XPath Axis following expressions and equivalent xpath expression.
+     */
+    @DataProvider(name = "followingXpath")
+    public Object[][] getFollowingXpathExpression() {
+        return new Object[][] {
+                {"/Customers/following::*", "/None"},
+                {"/Customers/Customer/following::Customer", "//Customer[@id != 'x1']"},
+                {"/Customers/Customer/following::foo:Customer", "//foo:Customer"},
+                {"/Customers/Customer[@id='x1']/following::Address", "/Customers/Customer[@id != 'x1']/Address"},
+                {"/Customers/Customer[@id='x1']/following::*", "/Customers/Customer[@id != 'x1']/descendant-or-self::* | /Customers/foo:Customer/descendant-or-self::*"},
+                {"/Customers/foo:Customer/foo:Address/following::*", "/Customers/foo:Customer/foo:Age | /Customers/foo:Customer/foo:ClubMember"},
+                {"/Customers/following-sibling::*", "/None"},
+                {"/Customers/Customer/following-sibling::Customer", "/Customers/Customer[@id != 'x1']"},
+                {"/Customers/Customer/following-sibling::foo:Customer", "/Customers/foo:Customer"},
+                {"/Customers/Customer[@id='x1']/Name/following-sibling::Address", "/Customers/Customer[@id='x1']/Address"},
+                {"/Customers/Customer/Name/following-sibling::Address", "/Customers//Address"},
+                {"/Customers/Customer[@id='x1']/following-sibling::*", "/Customers/Customer[@id != 'x1'] | /Customers/foo:Customer"},
+                {"/Customers/foo:Customer/foo:Address/following-sibling::*", "/Customers/foo:Customer/foo:Age | /Customers/foo:Customer/foo:ClubMember"}
+        };
```

Comment:
It would be good to add a position test. e.g.
`//Customer/following::Street[2]`
`//following::Street[2]`
`//Customer/following-sibling::Customer[2]`
`//following-sibling::Customer[2]`